### ISSUE TITLE
up_iter

### ISSUE THREAD

**Issue by douhongkun** [ucr-optml/AutoBalance#1]
First of all, thank you for your work, but I have some doubts when looking at the code. What does this part of back propagation do? It seems that the algorithm in the paper does not have this part
![image](https://user-images.githubusercontent.com/77845036/154825356-002c075d-46b3-42a6-91f7-9e7ed870ff1f.png)

**Comment by mc-nya** [comment]
Sorry for leaving that part in the code and make confusion. That part computes the direct gradient of validation loss w.r.t upper parameters, this part is used in our initial proof-of-concept where validation loss contains the hyper-parameter. However, in Autobalance, since validation loss doesn't contain hyper-parameters (e.g. dy,ly), it will not affect the optimization and this part can be removed.

**Comment by douhongkun** [comment]
I see，thank you
